src/modules/git_status.rs

```diff
@@ -170,15 +206,46 @@ fn get_repo_status(repository: &Repository) -> Result<Status, git2::Error> {
 
     let repo_file_statuses = repository.statuses(Some(&mut status_options))?;
 
-    // Statuses are stored as bitflags, so use BitOr to join them all into a single value
-    let repo_status: Status = repo_file_statuses.iter().map(|e| e.status()).collect();
-    if repo_status.is_empty() {
+    if repo_file_statuses.is_empty() {
         return Err(git2::Error::from_str("Repo has no status"));
     }
 
+    let repo_status: RepoStatus = repo_file_statuses.iter().fold(RepoStatus::default(), count);
+
     Ok(repo_status)
 }
 
+/// Increment RepoStatus counts for a given file status
+fn count(mut repo_status: RepoStatus, status_entry: StatusEntry) -> RepoStatus {
+    let status = status_entry.status();
+
+    if status.is_conflicted() {
+        repo_status.conflicted += 1;
+    }
+
+    if status.is_wt_deleted() || status.is_index_deleted() {
+        repo_status.deleted += 1;
+    }
+
+    if status.is_wt_renamed() || status.is_index_renamed() {
+        repo_status.renamed += 1;
+    }
+
+    if status.is_wt_modified() {
+        repo_status.modified += 1;
+    }
+
+    if status.is_index_modified() || status.is_index_new() {
+        repo_status.staged += 1;
+    }
+
+    if status.is_wt_new() {
+        repo_status.untracked += 1;
+    }
+
+    repo_status
+}
```

Comment:
Done, let me know there is a shorter / more elegant was to do this

Reply:
Mhh. I don't see why any of this *has* to be mutable. I think you could do
```
let repo_status = if status.is_conflicted() {
        RepoStatus { conflicted: repo_status.conflicted + 1, .. repo_status };
    } else {
        repo_status
};
```
but I guess this will result in copies.

Reply:
Thanks @bijancn. I didn't know about `..`. Since starship is focused on speed I imagine the mutable version is preferred?

Reply:
IDK. Personally, I think one should always benchmark and consider multi-threading / async when one uses speed as an argument but I'm also fine with leaving it as it is.

Reply:
@bijancn Made immutable. Some informal testing with 50,000 untracked files showed no measurable difference via mutable.

Reply:
@bijancn Made immutable. Some informal testing with 50,000 untracked files showed no measurable difference via mutable.

Reply:
Thats great news <3

Reply:
Idk if it looks better but what about using

```rust
RepoStatus {
    conflicted: repo_status.conflicted + i32::from(repo_status.conflicted()),
    deleted: ...,
    ...,
}
```

Reply:
@heyrict Can you provide more details on `repo_status.conflicted()`? I expected to see `status` used somewhere.

I experimented with what I thought was the most readable solution. Informal benchmarking shows no speed difference.
```rust
    let repo_status: RepoStatus = RepoStatus {
        conflicted: statuses.iter().filter(|s| is_conflicted(**s)).count(),
        deleted: statuses.iter().filter(|s| is_deleted(**s)).count(),
        renamed: statuses.iter().filter(|s| is_renamed(**s)).count(),
        // ...
    };
```
What do you think?

Reply:
Oh, sorry. It's a typo of `status.is_conflicted()`.

> I experimented with what I thought was the most readable solution...

That looks great 👍